packages/dev/core/src/Maths/math.vector.ts

```diff
@@ -2735,6 +2781,39 @@ export class Vector3 {
         Quaternion.RotationQuaternionFromAxisToRef(axis1, axis2, axis3, quat);
         quat.toEulerAnglesToRef(ref);
     }
+
+	/**
+	 * Gets the pitch, yaw, roll needed to change direction from one vector to another
+	 * @param vector0 defines the first vector
+	 * @param vector1 defines the second vector
+	 * @param offset defines the amount to offset the output rotation
+	 * @param result the rotation (pitch, yaw, roll) to apply
+	 * @returns the updated result in the form (pitch, yaw, roll)
+	 * Output roll will always be 0 + offset.z
+	 */
+	public static PitchYawRollForDirectionChangeToRef(vector0: DeepImmutable<Vector3>, vector1: DeepImmutable<Vector3>, result: DeepImmutable<Vector3>): Vector3{

```

Comment:
This does not work as specified.

In this PG https://playground.babylonjs.com/#ZUFVG7#6
even though the asym cone points along the given vector V there are two issues 

The current vector along which the asym cone points is the Y axis however this is not passed to the function it is the mesh position (a zero vector) noa direction that is passed and also without a correction of a rotation of -PI / 2 the cone points in the wrong direction. 

My recommendation is to use the following functions. Note as pointed out in https://forum.babylonjs.com/t/rotating-vector-to-align-with-other-vector/33776/9?u=johnk a little more work is needed.

Reply:
The intent of the function is not to take into account the current rotation of a mesh, only its position. 

The reason -pi/2 is added is because while the cone points upward by default, the actual rotation is in the direction of the box - not the cone's tip. Hence, when properly aligned to the target, the cone's tip would not face it, the box would.

In https://playground.babylonjs.com/#QYA3JC you use the following functions:
```js
const QuaternionFromTo = (fromVec, toVec) => {
    const normal = BABYLON.Vector3.Cross(fromVec, toVec);
    const angle = BABYLON.Vector3.GetAngleBetweenVectors(fromVec, toVec, normal);
    return BABYLON.Quaternion.RotationAxis(normal, angle);
}

const RotationFromTo = (fromVec, toVec) => {
    return QuaternionFromTo(fromVec, toVec).toEulerAngles();
}
```
Which seems to get the rotation from a rotation Vector3 to a position Vector3 - could you clarify the purpose of them?

Reply:
Also, I've attempted to use this with the below PG, but it does not appear to work properly (the box should face the same direction it is moving)
https://playground.babylonjs.com/#2I8Y3S#14

Reply:
> Would you confirm that its use is given an object moving in one vector direction to find the rotation in terms of pitch, yaw, roll needed to change the objects direction to move along a second vector direction?

2 days ago

Yes, that is what the function does, and that is a great explanation.

Today
The intent of the function is not to take into account the current rotation of a mesh, only its position.

@sebavan I cannot approve this PR. Ad I understood the intention there are too many issues to be resolved.

@dr-vortex and I seem to be working across each other. It is not enough to make a comment and say the conversation is resolved.

Reply:
For clarification:

> The current vector along which the asym cone points is the Y axis however this is not passed to the function it is the mesh position (a zero vector) not direction that is passed and also without a correction of a rotation of -PI / 2 the cone points in the wrong direction.

Why would the current direction of the cone be passed?

Reply:
Also, for my earlier comment
> Would you confirm that its use is given an object moving in one vector direction to find the rotation in terms of pitch, yaw, roll needed to change the objects direction to move along a second vector direction?
> Yes, that is what the function does, and that is a great explanation.

I clearly misread the question.

I'm sorry for all the confusion caused, I originally and continue to intend for the function to be based on an origin position rather than an origin direction.

Reply:
> Also, I've attempted to use this with the below PG, but it does not appear to work properly (the box should face the same direction it is moving)
> https://playground.babylonjs.com/#2I8Y3S#14

It works here https://playground.babylonjs.com/#1PX9G0#61

Reply:
Interesting...
https://playground.babylonjs.com/#1PX9G0#63

I would expect the smaller box with a position relative to the larger box of Vector3.Forward() to be in the same direction of travel, though it is off to the side. Maybe this is just an oversight?